tests/test_new_return_types_qnode.py

```diff
@@ -0,0 +1,1665 @@
+        assert len(res) == 2
+        for elem in res:
+            assert isinstance(elem, jax.numpy.ndarray)
+            assert elem.shape == (2, 3)
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_meas_autograd(self, device):
+        """Return Jacobian of multiple measurements."""
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev)
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.probs(wires=[0, 1]), qml.vn_entropy(wires=1)
+
+        x = qml.numpy.array([0.1, 0.2, 0.3], requires_grad=True)
+        res = qml.jacobian(circuit)(x)
+
+        qml.disable_return()
+
+        assert isinstance(res, np.ndarray)
+        assert res.shape == (6, 3)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_meas_torch(self, device):
+        """Return Jacobian of multiple measurements."""
+        import torch
+
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="torch")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.probs(wires=[0, 1]), qml.vn_entropy(wires=1)
+
+        x = torch.tensor([0.1, 0.2, 0.3])
+
+        res = torch.autograd.functional.jacobian(circuit, x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tuple)
+        assert len(res) == 3
+        for i, elem in enumerate(res):
+            assert isinstance(elem, torch.Tensor)
+            if i == 0:
+                assert elem.shape == (3,)
+            elif i == 1:
+                assert elem.shape == (4, 3)
+            elif i == 2:
+                assert elem.shape == (3,)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_meas_tf(self, device):
+        """Return Jacobian of multiple measurements."""
+        import tensorflow as tf
+
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return (
+                qml.expval(qml.PauliZ(wires=0)),
+                qml.probs(wires=[0, 1]),
+                qml.var(qml.PauliZ(wires=0)),
+            )
+
+        x = tf.Variable([0.1, 0.2, 0.3])
+
+        with tf.GradientTape() as tape:
+            out = circuit(x)
+
+        res = tape.jacobian(out, x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (6, 3)
+
+    @pytest.mark.jax
```

Comment:
Some interfaces (e.g., JAX, but not Autograd) support complex-valued function differentiation too with backprop. Maybe good to have a case for those too.

Reply:
Do you have an example of qnode returning complex which is not state? Otherwise I don't see a practical case